helm-system-packages-brew.el

```diff
@@ -0,0 +1,214 @@
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;; Helm UI for Mac OS' homebrew.
+
+;; TODO
+;; (1) Implement `show-dependencies' function
+;; (2) Show packages explicitly installed (function `helm-system-packages-brew-list-explicit' already exists)
+;; (3) Implement `find-files'
+;; (4) Keymap
+
+;;; Code:
+(require 'helm)
+(require 'json)
+(require 'helm-system-packages)
+
+(defun helm-system-packages-brew-transformer (packages)
+  (let (res (pkglist (reverse packages)))
+    (dolist (p pkglist res)
+      (let ((face (cdr (assoc (helm-system-packages-extract-name p) helm-system-packages--display-lists))))
+        (cond
+         ((and (not face) helm-system-packages--show-uninstalled-p)
+               (push p res)))))))
+
+(defun helm-system-packages-brew-list-explicit ()
+  "List explicitly installed packages."
+  (split-string (with-temp-buffer
+                  (call-process "brew" nil t nil "list")
+                  (buffer-string))))
+
+(defcustom helm-system-packages-brew-column-width 40
+  "Column at which descriptions are aligned, excluding a double-space gap.
+If nil, then use `helm-system-package-column-width'."
+  :group 'helm-system-packages
+  :type 'integer)
+
+(defun helm-system-packages-brew-cache ()
+  "Cache all package names with descriptions.
+LOCAL-PACKAGES and GROUPS are lists of strings.
+Return (NAMES . DESCRIPTIONS), a cons of two strings."
+  ;; We build both caches at the same time.  We could also build just-in-time, but
+  ;; benchmarks show that it only saves less than 20% when building one cache.
+  (let (names descriptions)
+    (setq descriptions
+          (with-temp-buffer
+            ;; TODO: Possible optimization: Output directly in Elisp?
+            (let ((format-string (format "%%-%dn  %%d" helm-system-packages-column-width)))
+              (call-process "brew" nil '(t nil) nil "desc" "-s" "" )
+	      (buffer-string))))
+    (setq names
+	  (replace-regexp-in-string ":.*" "" descriptions))      
+    (setq descriptions (mapconcat (lambda (package-from-list)
+				    (let* ((pkg (split-string package-from-list ": "))
+					   (name (car pkg))
+					   (desc (car (cdr pkg)))
+					   (format-string (format "%%-%ds  %%s" helm-system-packages-column-width)))
+				    (format format-string name desc)))
+				  (split-string descriptions"\n") "\n" ))
+  (cons names descriptions)))
+
+(defun helm-system-packages-brew-init ()
+  "Cache package lists and create Helm buffer."
+  (unless (and helm-system-packages--names helm-system-packages--descriptions)
+    (helm-system-packages-brew-refresh))
+  ;; TODO: We should only create the buffer if it does not already exist.
+  ;; On the other hand, we need to be able to override the package list.
+  ;; (unless (helm-candidate-buffer) ...
+  (helm-init-candidates-in-buffer
+      'global
+    (if helm-system-packages-show-descriptions-p
+        helm-system-packages--descriptions
+      helm-system-packages--names)))
```

Comment:
This function can now be removed altogether: use `helm-system-packages-init` instead.